CONTRIBUTING.md

```diff
@@ -0,0 +1,157 @@
+
+## Submission Guidelines
+
+### Submitting an Issue
+
+Before you submit an issue, please check the issue tracker to see if a similar issue already exists. The discussions there may provide workarounds.
+
+For us to address and fix a bug, we need to reproduce it. Thus when submitting a bug report, we will ask for a minimal reproduction scenario using a repository or [Gist](https://gist.github.com/). Providing a live, reproducible scenario helps us understand the issue better. Information to include:
+
+- The version of the document-service you are using.
+- Any third-party libraries and their versions.
+- A use-case that demonstrates the issue.
+
+Without a minimal reproduction, we may need to close the issue due to insufficient information.
+
+You can file new issues using our [new issue form](https://github.com/OsmosysSoftware/document-service/issues/new).
+
+### Submitting a Pull Request (PR)
+
+Before submitting a Pull Request (PR), please follow these guidelines:
+
+1. Search GitHub [Pull Requests](https://github.com/OsmosysSoftware/document-service/pulls) to ensure there is no open or closed PR 
+   related to your submission.
+2. Fork this repository.
+3. Make your changes in a new Git branch.
+   ```csharp
+   git checkout -b my-fix-branch main
+   ```
+4. Create your patch, **including appropriate test cases**.
+5. Follow our coding rules, which is mentioned below.
+6. Run the full document-service test suite to ensure that all tests pass.
+7. Commit your changes using a descriptive commit message that adheres to our commit message conventions.
+   ```csharp
+   git commit -a
+   ```
+   Note: the optional commit -a command line option will automatically "add" and "rm" edited files.
+8. Push your branch to your GitHub repository.
+   ```csharp
+   git push origin my-fix-branch
+   ```
+9. Send a pull request to the `document-service:main`.
+
+- <strong style="color:black">�</strong> **If we suggest changes, then:**
+  - � Make the required updates.
+  - � Re-run the Osmo-Notify test suites to ensure tests are still passing.
+  - � Rebase your branch and force push to your GitHub repository. This will update your Pull Request.
+
+  That's it! Thank you for your contribution!
+
+## Feature Requests
+
+You can request new features by submitting an issue to our GitHub repository. If you intend to implement a new feature, start by proposing it through 
+an issue. Major features require discussion, so please prefix your proposal with [discussion], for example, "[discussion]: your feature idea.
+" Smaller features can be crafted and directly submitted as a Pull Request.
+
+## Coding Rules
+
+To ensure consistency throughout the source code, follow these rules as you work on the project:
+
+- All features or bug fixes must be tested by one or more specs (unit-tests).
+
+- We follow the [visual studio coding standards](https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/coding-style/coding-conventions), 
```

Comment:
You can link to the Omsosys coding standards for .NET/C#
https://github.com/OsmosysSoftware/dev-standards/blob/main/coding-standards/dotnet.md

Reply:
Updated